test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -144,22 +151,223 @@ contract LoansForkTest is LoansTestBase {
+        pair.escrowNFT.withdrawReleased(loanBefore.escrowId);
+        vm.stopPrank();
+
+        assertEq(
+            pair.underlying.balanceOf(escrowSupplier) - escrowSupplierBefore,
+            underlyingAmount + interestFee + lateFee
+        );
+    }
+
+    function testRollEscrowLoanBetweenSuppliers() public {
+        // Create first provider and escrow offers
+        (uint offerId1, uint escrowOfferId1) = _createEscrowOffers();
+        uint interestFee1 = pair.escrowNFT.interestFee(escrowOfferId1, underlyingAmount);
+
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+        (uint loanId, uint providerId, uint loanAmount) = _executeEscrowLoan(offerId1, escrowOfferId1);
+
+        // User has paid underlyingAmount + interestFee1 at this point
+        assertEq(userUnderlyingBefore - pair.underlying.balanceOf(user), underlyingAmount + interestFee1);
+
+        uint escrowSupplier1Before = pair.underlying.balanceOf(escrowSupplier);
+
+        // Create second escrow supplier
+        address escrowSupplier2 = makeAddr("escrowSupplier2");
+        deal(address(underlying), escrowSupplier2, bigUnderlyingAmount);
+
+        // Skip half the duration to test partial fees
+        skip(pair.durations[0] / 2);
+
+        // Get exact refund amount from contract
+        (, uint toLoans,) = pair.escrowNFT.previewRelease(loanId, 0);
+
+        // Create second escrow supplier's offer
+        vm.startPrank(escrowSupplier2);
+        pair.underlying.approve(address(pair.escrowNFT), underlyingAmount);
+        uint escrowOfferId2 = pair.escrowNFT.createOffer(
+            underlyingAmount,
+            pair.durations[0],
+            interestAPR,
+            maxGracePeriod,
+            lateFeeAPR,
+            0 // minEscrow
+        );
+        vm.stopPrank();
+
+        // Create roll offer using existing provider position
+        uint rollOfferId = createRollOffer(pair, provider, loanId, providerId, rollFee, rollDeltaFactor);
+
+        uint newEscrowFee = pair.escrowNFT.interestFee(escrowOfferId2, underlyingAmount);
+        uint userUnderlyingBeforeRoll = pair.underlying.balanceOf(user);
+
+        vm.startPrank(user);
+        pair.cashAsset.approve(address(pair.loansContract), loanAmount);
+        pair.underlying.approve(address(pair.loansContract), newEscrowFee);
+        (uint newLoanId, uint newLoanAmount,) = pair.loansContract.rollLoan(
+            loanId,
+            ILoansNFT.RollOffer(pair.rollsContract, rollOfferId),
+            -1000e6,
+            escrowOfferId2,
+            newEscrowFee
+        );
+        vm.stopPrank();
+
+        // Execute withdrawal for first supplier
+        vm.startPrank(escrowSupplier);
+        pair.escrowNFT.withdrawReleased(loanId);
+        vm.stopPrank();
+
+        // Verify first supplier got partial fees using exact toLoans amount
+        // we use Ge because of rounding (seeing 1 wei difference 237823439879 != 237823439878)
```

Comment:
it shouldn't be toLoans (this is what loans receive), it should be `withdrawal` (first return value from `previewRelease`)